src/sentry/sentry_metrics/indexer/models.py

```diff
@@ -0,0 +1,30 @@
+from typing import Any
+
+from django.db import connections, models, router
+
+from sentry.db.models import Model
+from sentry.db.models.fields.bounded import BoundedBigIntegerField
+
+
+class MetricsKeyIndexer(Model):  # type: ignore
+    __include_in_export__ = False
+
+    organization_id = BoundedBigIntegerField()
+    string = models.CharField(max_length=200)
+
+    class Meta:
+        db_table = "sentry_metricskeyindexer"
+        app_label = "sentry"
+        constraints = [
+            models.UniqueConstraint(fields=["organization_id", "string"], name="unique_org_string"),
+        ]
+
+    @classmethod
+    def get_next_values(cls, num: int) -> Any:
+        using = router.db_for_write(cls)
+        connection = connections[using].cursor()
+
+        connection.execute(
+            "SELECT nextval('sentry_metricskeyindexer_id_seq') from generate_series(1,%s)", [num]
+        )
+        return connection.fetchall()
```

Comment:
I don't have a lot of context on this project, how will we use the values from this sequence?

It vaguely looks like you want to reserve a range of ids, and then use those ids later on to create new rows. Is that the general idea?

Reply:
@wedamija Sorry for not giving enough context in the PR description, I can go back an update in a bit but yeah:

> It vaguely looks like you want to reserve a range of ids, and then use those ids later on to create new rows. Is that the general idea?

Thats basically it. Eventually we want to have postgres be off the critical path, but in order to do that we need to know the ids ahead of time. What I am unsure about is what kind of ranges we are talking, is it 100, 1000, 10000? Since this metrics indexer will be used by metrics names, tag keys, and tag values, it could be a lot of writes for high cardinality tags

Reply:
Looks good for now. Once we know how many ids we're allocation per second we can decide whether we need to do something more complex here. 

I'm not sure if there's a performance hit to calling `nextval` 10k times, as opposed to doing something like https://www.depesz.com/2008/03/20/getting-multiple-values-from-sequences/. Something we can possibly benchmark in the future.

Reply:
I bet it will be desirable to avoid calling nextval 10k times (that would be 10k writes I believe). We may not strictly need a sequence at that point but just a counter.
I think we could discuss the solutions then. At this point I am not sure it is very useful to have this method at this stage.  Probably better removing it for now in case somebody decided to start depending on it for some reason.